dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/appmanager/DefaultAppManager.java

```diff
@@ -395,6 +395,22 @@ public Resource getAppResource(App app, String pageName) throws IOException {
     return getAppStorageServiceByApp(app).getAppResource(app, pageName);
   }
 
+  /**
+   * @param resource resource to check content length
+   * @return the content length or -1 (unknown size) if exception caught
+   */
+  @Override
+  public int getUriContentLength(Resource resource) {
+    try {
+      URLConnection urlConnection = resource.getURL().openConnection();
+      return urlConnection.getContentLength();
```

Comment:
There is a `getContentLong()` method. Are we using `int` on purpose ?

Reply:
As with most things, you're hoping the initial implementation chose `int` for a reason :) 
I'd hope the App would never need to return a resource bigger than `Integer.MAX_VALUE` (~2GB)